### PR TITLE
Add aten mkldnn batch_norm operator

### PR DESCRIPTION
**Pull request by bddppq**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19633 Add is_mkldnn to at::Tensor&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15053320/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19204 Add aten mkldnn conv2d operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14857513/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19205 Add aten mkldnn ops: relu, max_pool2d and avg_pool2d&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14850598/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19206 Add aten mkldnn batch_norm operator**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14887205/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19207 Add aten mkldnn add operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14889477/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19209 Add aten mkldnn view operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14894545/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19210 Add aten mkldnn linear operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14901641/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19648 Adjust resnext run script&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14889481/)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/19206

Differential Revision: [D14887205](https://our.internmc.facebook.com/intern/diff/D14887205/)

### CODE REVIEW

File: aten/src/ATen/native/mkldnn/Normalization.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    double eps) {
+  ideep::tensor& x = itensor_from_mkldnn(input);
+  ideep::tensor& w = itensor_from_mkldnn(weight);
+  ideep::tensor& b = itensor_from_mkldnn(bias);
+  ideep::tensor& m = itensor_from_mkldnn(running_mean);
+  ideep::tensor& v = itensor_from_mkldnn(running_var);
+
+  ideep::tensor y;
+
+  if (train) {
```
Comment (XiaobingSuper) [inline aten/src/ATen/native/mkldnn/Normalization.cpp]: @bddppq , I think there should check whether running status is used for training and infernce, i.e. running_mean and running_var are defined or not.  Another suggestion is that  mkldnn only support batchnorm2d and batchnorm3d, so you shoud make some checks when call mkldnn, perhaps you can see the  [code](https://github.com/XiaobingSuper/pytorch/commit/ed404f4b748a7cf38e9f466743d2d2bc6aedf003)  as reference. thanks!

Comment (dzhulgakov) [reply aten/src/ATen/native/mkldnn/Normalization.cpp]: would mkldnn throw with nice message that it's not supported? in this case the input is already mkldnn tensor so it's better to fail if something is not supported

**Review by dzhulgakov — APPROVED** [review]

File: aten/src/ATen/native/mkldnn/Normalization.cpp (hunk 1 above)
Comment (jgong5) [reply aten/src/ATen/native/mkldnn/Normalization.cpp]: @dzhulgakov I'm afraid not. :-(   How about we add an assertion here to guarantee 2d or 3d batchnorm here?

Comment (dzhulgakov) [inline aten/src/ATen/native/mkldnn/Normalization.cpp]: This training support is incomplete. Derivatives.yaml calls native_batch_norm_backward and that one doesn't know what to do with mkldnn tensors. Thus I'd say either assert that train=false or implement native_batch_norm_backward for consistency

Comment (bddppq) [reply aten/src/ATen/native/mkldnn/Normalization.cpp]: Makes senses will add an assert
